lib/activerecord-tenant-level-security/schema_statements.rb

```diff
@@ -1,22 +1,22 @@
 module TenantLevelSecurity
   module SchemaStatements
-    def create_policy(table_name)
+    def create_policy(table_name, partition_key: 'tenant_id')
       execute <<~SQL
         ALTER TABLE #{table_name} ENABLE ROW LEVEL SECURITY;
         ALTER TABLE #{table_name} FORCE ROW LEVEL SECURITY;
       SQL
-      tenant_id_data_type = get_tenant_id_data_type(table_name)
+      tenant_id_data_type = get_tenant_id_data_type(table_name, partition_key)
       execute <<~SQL
         CREATE POLICY tenant_policy ON #{table_name}
           AS PERMISSIVE
           FOR ALL
           TO PUBLIC
-          USING (tenant_id = NULLIF(current_setting('tenant_level_security.tenant_id'), '')::#{tenant_id_data_type})
-          WITH CHECK (tenant_id = NULLIF(current_setting('tenant_level_security.tenant_id'), '')::#{tenant_id_data_type})
+          USING (#{partition_key} = NULLIF(current_setting('tenant_level_security.tenant_id'), '')::#{tenant_id_data_type})
+          WITH CHECK (#{partition_key} = NULLIF(current_setting('tenant_level_security.tenant_id'), '')::#{tenant_id_data_type})
```

Comment:
As you say, it would certainly be better to use ActiveRecord's functions to absorb the differences.
I fixed it in this commit: https://github.com/kufu/activerecord-tenant-level-security/pull/22/commits/40989f89aaf581e3280f1e155072d8541a534023